src/main/java/net/helix/pendulum/crypto/Merkle.java

```diff
@@ -55,6 +60,44 @@
     }
 
 
+    public static List<byte[]> buildMerkleTransactionTree(List<Hash> leaves, Hash milestoneHash, Hash address){
+        if (leaves.isEmpty()) {
+            leaves.add(Hash.NULL_HASH);
+        }
+        byte[] buffer;
+        Sponge sha3 = SpongeFactory.create(SpongeFactory.Mode.S256);
+        int depth = (int) Math.ceil(Math.sqrt(leaves.size()));
+        int row = 1;
+        List<byte[]> virtualTransactionList = new ArrayList<byte[]>();
+        while (leaves.size() > 1) {
+            List<Hash> nextKeys = Arrays.asList(new Hash[(leaves.size() / 2)]);
+            for (int i = 0; i < nextKeys.size(); i++) {
+                if (areLeavesNull(leaves, i)) continue;
+                sha3.reset();
+                Hash k1 = leaves.get(i * 2);
+                Hash k2 = leaves.get(i * 2 + 1);
+                buffer = computeParentHash(sha3, k1, k2);
+                Hash parentHash = HashFactory.TRANSACTION.create(buffer);
+                nextKeys.set(i, parentHash);
+               // if( i == 0) add only one virtual transaction for each level
+                virtualTransactionList.add(createVirtualTransaction(k1, k2, row, milestoneHash, address, parentHash));
+            }
+            leaves = nextKeys;
+            row++;
+        }
+        return virtualTransactionList;
+    }
+
+    private static byte[] computeParentHash(Sponge sha3, Hash k1, Hash k2) {
+        byte[] buffer;
+        buffer = Arrays.copyOfRange(k1 == null ? Hex.decode("0000000000000000000000000000000000000000000000000000000000000000") : k1.bytes(), 0, 32);
+        sha3.absorb(buffer, 0, buffer.length);
+        buffer = Arrays.copyOfRange(k2 == null ? Hex.decode("0000000000000000000000000000000000000000000000000000000000000000") : k2.bytes(), 0, 32);
```

Comment:
Perhaps turn `Hex.decode('00000....')` into a constant somewhere in utils? Same for the whole routine 
```
Arrays.copyOfRange(k2 == null ? Hex.decode("0000000000000000000000000000000000000000000000000000000000000000") : k2.bytes(), 0, 32);
``` 
Looks like quite a used routine across the code